plugins/inventory/vmware_vm_inventory.py

```diff
@@ -69,67 +69,69 @@
+                       'config.name', 'config.uuid' ,'guest.hostName', 'guest.ipAddress',
                        'guest.guestId', 'guest.guestState', 'runtime.maxMemoryUsage',
-                       'customValue'
+                       'customValue', 'summary.runtime.powerState', 'config.guestId'
                        ]
             version_added: "2.9"
-        datacenters:
+        resources:
             description:
-            - A list of names of the datacenters.
+            - A list of resources to limit search scope.
+            - Each resource item is represented by exactly one C('vim_type_snake_case):C(list of resource names) pair and optional nested I(resources)
+            - Key name is based on snake case of a vim type name; e.g C(host_system) correspond to C(vim.HostSystem)
+            - See  L(VIM Types,https://pubs.vmware.com/vi-sdk/visdk250/ReferenceGuide/index-mo_types.html)
             version_added: "2.10"
             required: False
             type: list
-        clusters:
+            default: []
+        with_path:
             description:
-            - A list of names of the clusters.
+            - Include virtual machines path.
+            - Set this option to a string value to replace root name from I('Datacenters').
+            default: False
+            type: bool
             version_added: "2.10"
-            required: False
-            type: list
-        folders:
+        hostnames:
             description:
-            - A list of folder name where VM reside.
-            - Folder that contains the virtual machine for the virtual machine to match the filter.
-            - This is not absolute or relative path to virtual machine but a single folder name.
-            version_added: "2.10"
-            required: False
+                - A list of templates in order of precedence to compose inventory_hostname.
+                - Ignores template if it resulted in an empty string or None value.
+                - You can use property specified in I(properties) as variables in template.
             type: list
-        esxi_hostsystems:
+            default: ['config.name + "_" + config.uuid']
+            version_added: "2.10"
+        with_nested_properties:
             description:
-            - A list of ESXi host systems.
+                - This option transform flatten properties name to nested dictionary.
+            type: bool
+            default: False
             version_added: "2.10"
-            required: False
-            type: list
-        resource_pools:
+        with_sanitized_property_name:
             description:
-            - A list of resource pools.
+                - This option allows property name sanitization to create safe property names for use in Ansible.
+                - Also, transforms property name to snake case.
+            type: bool
+            default: False
             version_added: "2.10"
-            required: False
+        keyed_groups:
```

Comment:
isn't this redundant from what is included by extending the constructed doc_fragment?

Reply:
This to set default value to `keyed_groups` to behave similarly to 2.9.